src/docs/guides/23-how-to-use-custom-pricing/index.md

```diff
@@ -0,0 +1,793 @@
+- **Seasonal pricing.** Examples: weekends cost more than weekdays,
+  summers cost more than winters, and so on.
+- **Quantity discount.** An example: booking a room for two people costs
+  $100, and
+  after that each additional person costs $20 extra.
+
+All of these can be achieved by storing pricing information in the
+[extended data](/references/extended-data/) of the listings, and then
+adjusting your [transaction process](/background/transaction-process/)
+by adding a new action called set-line-items-and-total. As the name
+suggests, this action allows you to set any number of line items you
+need (based on the price information stored in extended data and the
+choices the customer makes when initiating the booking) and calculate
+the total price of the transaction based on them.
+
+This guide walks you through the process of using the
+set-line-items-and-total action by using the example of a common pricing
+adjustment: adding the ability to charge an optional cleaning fee on top
+of the regular nightly price of the accommodation.
+
+## 1. Process change
+
+To get started, the transaction process needs to be updated to support
+custom pricing. The requierd change is to add and action called
+`set-line-items-and-total` to all the transitions that create a booking.
+For this, you need to
+[contact Sharetribe support](mailto:flex-support@sharetribe.com) and ask
+them to make the process adjustment for you. In this guide, we're
+assuming that the `set-line-items-and-total` action has been added to
+the `transition/request` and `transition/request-after-enquiry`
+transitions in the default transaction process.
+
+Note: `set-line-items-and-total` cannot be combined with other pricing
+actions that rely on the booking length or the quantity of units. In
+order to take custom pricing into use, price calculations that rely on
+booking length or quantity of units also need to be converted into using
+custom pricing. Commissions and price negatiation however don't require
+any changes as they can be combined with custom pricing.
+
+## 2. Listing extended data
+
+Custom pricing can be based on a lot of variables but one practical way
+to build it is to base it on information stored as extended data in
+listings. See the
+[How to extend listing data in FTW](/guides/how-to-extend-listing-data-in-ftw/)
+guide to read how to extend the listing data model with extended data.
+
+## 3. Change booking creation to build line items
+
+The first step to do in FTW is to modify the current booking creation so
+that custom pricing is used instead of the nightly pricing. For custom
+pricing, the booking is not constructed in the API based on
+`bookingStart` and `bookinEnd` but it's based on line items that the
+client sends in the booking initiation request.
+
+Instead of the just listing ID, now we're going to pass the actual line
+items in the transaction initiate request. To achieve this
+`CheckoutPage` has to be modified a bit. Add the following after the
```

Comment:
I'd change this a bit: ~~after the~~ to the.

It's up to customizers to deal with import order, but since it can create cyclic dependency loops it's usually best to import utility files before components and other complex creatures that might also fetch same utility files.